File: Data Structures & Algorithms/anagram-groups/submission-3.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
         map<string,vector<string>> anagrams;
    vector<vector<string>> res;
    for (auto str : strs)
    {
        string temp= str;
        sort(str.begin(),str.end());
        if (anagrams.contains(str))
            anagrams[str].push_back(temp);
        else
            anagrams[str]={temp};
    }
    for (auto [str,arr] : anagrams)
        res.push_back(arr);
    return res;
    }
};

File: Data Structures & Algorithms/string-encode-and-decode/submission-3.cs
public class Solution {

    public string Encode(IList<string> strs) {
        string res = "";
        foreach(string str in strs)
        {
            res+=str+'.';
        }
        return res;
    }

    public List<string> Decode(string s) {
        List<string> res= new List<string>();
        string temp="";
        foreach(char c in s)
        {
            if(c=='.')
            {
                res.Add(temp);
                temp="";
            }
            else 
                temp+=c;
        }
        return res;
   }
}
